File: Third_Year/DS/Practicas/P1/Filtros/FiltroIva.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

#include "FiltroIva.h"
#include "Alquiler.h"

void FiltroIva::ejecutar(Alquiler &alq){
    float precio = alq.getPrecio();
    precio += precio*IVA;
    alq.setPrecio(precio);
}

File: Third_Year/DS/Practicas/P4/codigo_flutter/FiltroIVA.dart
import 'Filtro.dart';
import 'Alquiler.dart';

class FiltroIVA implements Filtro {
  double IVA = 0.21;
  void ejecutar(Alquiler alq) {
    double precio = alq.getPrecio();
    precio = precio + precio * IVA;
    alq.setPrecio(precio);
  }
  FiltroIva(){}
}
